File: InterviewPreparationKit/2DArray.py
#!/bin/python3
# https://www.hackerrank.com/challenges/2d-array/problem
import math
import os
import random
import re
import sys

# Complete the hourglassSum function below.
def hourglassSum(arr):
    maximum = float('-inf')

    for i in range(16):
        first = arr[int(i / 4)][i % 4]
        second = arr[int(i / 4)][(i % 4) + 1]
        third = arr[int(i / 4)][(i % 4) + 2]
        fourth = arr[int(i / 4) + 1][(i % 4) + 1]
        fifth = arr[int(i / 4) + 2][i % 4]
        sixth = arr[int(i / 4) + 2][(i % 4) + 1]
        seventh = arr[int(i / 4) + 2][(i % 4) + 2]

        total = first + second + third + fourth + fifth + sixth + seventh

        if total > maximum:
            maximum = total
    
    return maximum

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    arr = []

    for _ in range(6):
        arr.append(list(map(int, input().rstrip().split())))

    result = hourglassSum(arr)

    fptr.write(str(result) + '\n')

    fptr.close()

File: InterviewPreparationKit/2DArray.java
// https://www.hackerrank.com/challenges/2d-array/problem
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Solution {

    // Complete the hourglassSum function below.
    static int hourglassSum(int[][] arr) {
        int max = Integer.MIN_VALUE;

        for (int i = 0; i < 16; i++) {
            int first = arr[i / 4][i % 4];
            int second = arr[i / 4][(i % 4) + 1];
            int third = arr[i / 4][(i % 4) + 2];
            int fourth = arr[(i / 4) + 1][(i % 4) + 1];
            int fifth = arr[(i / 4) + 2][i % 4];
            int sixth = arr[(i / 4) + 2][(i % 4) + 1];
            int seventh = arr[(i / 4) + 2][(i % 4) + 2];

            int sum = first + second + third + fourth + fifth + sixth + seventh;
            if (sum > max) {
                max = sum;
            }
        }

        return max;
    }

    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) throws IOException {
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int[][] arr = new int[6][6];

        for (int i = 0; i < 6; i++) {
            String[] arrRowItems = scanner.nextLine().split(" ");
            scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

            for (int j = 0; j < 6; j++) {
                int arrItem = Integer.parseInt(arrRowItems[j]);
                arr[i][j] = arrItem;
            }
        }

        int result = hourglassSum(arr);

        bufferedWriter.write(String.valueOf(result));
        bufferedWriter.newLine();

        bufferedWriter.close();

        scanner.close();
    }
}
